public/app/features/scenes/components/layout/SceneGridLayout.tsx

```diff
@@ -0,0 +1,403 @@
+        }
+      },
+    });
+  }
+
+  onCollapseToggle = () => {
+    if (!this.state.isCollapsible) {
+      return;
+    }
+    const layout = this.parent;
+
+    if (!layout || !(layout instanceof SceneGridLayout)) {
+      throw new Error('SceneGridRow must be a child of SceneGridLayout');
+    }
+
+    const { isCollapsed, size } = this.state;
+    if (!size) {
+      return;
+    }
+
+    if (layout) {
+      if (isCollapsed) {
+        this.setState({ isCollapsed: false, isResizable: true, size: { ...size, height: this._originalHeight } });
+      } else {
+        this.setState({ isCollapsed: true, isResizable: false, size: { ...size, height: 1 } });
+      }
+      layout.updateLayout();
+    }
+  };
+}
+
+function SceneGridRowRenderer({ model }: SceneComponentProps<SceneGridRow>) {
+  const styles = useStyles2(getSceneGridRowStyles);
+  const { isCollapsible, isCollapsed, title, ...state } = model.useState();
+  const layout = model.getLayout();
+  const isDraggable = layout.state.isDraggable ? state.isDraggable : false;
+  const dragHandle = <SceneDragHandle layoutKey={layout.state.key!} />;
+
+  return (
+    <div className={styles.row}>
+      <div className={cx(styles.rowHeader, isCollapsed && styles.rowHeaderCollapsed)}>
+        <div onClick={model.onCollapseToggle} className={styles.rowTitleWrapper}>
+          {isCollapsible && <Icon name={isCollapsed ? 'angle-right' : 'angle-down'} />}
+          <span className={styles.rowTitle}>{title}</span>
+        </div>
+        {isDraggable && <div>{dragHandle}</div>}
+      </div>
+
+      {!isCollapsed && (
+        <div style={{ display: 'flex', flexGrow: 1, height: 'calc(100%-30px)', width: '100%' }}>
+          {model.state.children.map((child) => {
+            return <child.Component key={child.state.key} model={child} />;
+          })}
+        </div>
+      )}
+    </div>
+  );
+}
+
+const getSceneGridRowStyles = (theme: GrafanaTheme2) => {
+  return {
+    row: css({
+      width: '100%',
+      height: '100%',
+      position: 'relative',
+      zIndex: 0,
+      display: 'flex',
+      flexDirection: 'column',
+    }),
+    rowHeader: css({
+      width: '100%',
+      height: '30px',
+      display: 'flex',
+      justifyContent: 'space-between',
+      marginBottom: '8px',
+      border: `1px solid transparent`,
+    }),
+    rowTitleWrapper: css({
+      display: 'flex',
+      alignItems: 'center',
+      cursor: 'pointer',
+    }),
+    rowHeaderCollapsed: css({
+      marginBottom: '0px',
+      background: theme.colors.background.primary,
+      border: `1px solid ${theme.colors.border.weak}`,
+      borderRadius: theme.shape.borderRadius(1),
+    }),
+    rowTitle: css({
+      fontSize: theme.typography.h6.fontSize,
+      fontWeight: theme.typography.h6.fontWeight,
+    }),
+  };
+};
+
+// Source: https://github.com/metabase/metabase/blob/master/frontend/src/metabase/dashboard/components/grid/utils.js#L28
```

Comment:
nit: if we're gonna have a line indicating the source, prefer using a stable link (commit hash) rather than `master`